### ISSUE TITLE
Maybe a incorrect error  with pkg.exports["."].browser.types

### ISSUE THREAD

**Issue by prisis** [publint/publint#140]
Hey :)

im getting this error

```
Errors:
      1. pkg.exports["."].browser.types types is interpreted as ESM when resolving with the "require" condition. This causes the types to only work when dynamically importing the package, even though the package exports CJS. Consider splitting out two "types" conditions for "import" and "require", and use the .cts extension, e.g. pkg.exports["."].browser.require.types: "./dist/index.browser.d.cts"
```

for this package.json

```
"sideEffects": false,
    "type": "module",
    "exports": {
        ".": {
            "edge-light": {
                "types": "./dist/is-color-supported.edge-light.d.mts",
                "default": "./dist/is-color-supported.edge-light.mjs"
            },
            "browser": {
                "types": "./dist/is-color-supported.browser.d.mts",
                "default": "./dist/is-color-supported.browser.mjs"
            },
            "require": {
                "types": "./dist/is-color-supported.server.d.cts",
                "default": "./dist/is-color-supported.server.cjs"
            },
            "import": {
                "types": "./dist/is-color-supported.server.d.mts",
                "default": "./dist/is-color-supported.server.mjs"
            }
        },
        "./browser": {
            "require": {
                "types": "./dist/is-color-supported.browser.d.cts",
                "default": "./dist/is-color-supported.browser.cjs"
            },
            "import": {
                "types": "./dist/is-color-supported.browser.d.mts",
                "default": "./dist/is-color-supported.browser.mjs"
            }
        },
        "./edge-light": {
            "require": {
                "types": "./dist/is-color-supported.edge-light.d.cts",
                "default": "./dist/is-color-supported.edge-light.cjs"
            },
            "import": {
                "types": "./dist/is-color-supported.edge-light.d.mts",
                "default": "./dist/is-color-supported.edge-light.mjs"
            }
        },
        "./package.json": "./package.json"
    },
    "main": "dist/is-color-supported.server.cjs",
    "module": "dist/is-color-supported.server.mjs",
    "browser": "./dist/is-color-supported.browser.mjs",
    "types": "dist/is-color-supported.server.d.ts",
```

The problem is that i need to follow the order "edge-light" and "browser" in "." to have the correct resolve in next.js... 

So is there a option to ignore this error or is this maybe a bug most bundler will use the right key
